src/Network/Pusher.hs

```diff
@@ -179,13 +181,17 @@ users pusher chan =
     requestParams <- liftIO $ Pusher.mkUsersRequest pusher chan <$> getTime
     HTTP.get (pusherConnectionManager pusher) requestParams
 
--- Handle webhooks asynchronously given their AppKey matches the one in our Pusher credential
+-- Handle webhooks given their AppKey matches the one in our Pusher credential
 -- and the WebhookEv is encrypted by the matching AppSecret.
 handleWebhooks
   :: MonadIO m
-  => Pusher -> (UTCTime -> WebhookEv -> IO ()) -> m ThreadId
-handleWebhooks pusher webhookF =
-  handleWebhooks'
+  => Pusher
+  -> m req
+  -> (req -> Maybe ([(B.ByteString,B.ByteString)],B.ByteString))
+  -> (UTCTime -> WebhookEv -> m ())
+  -> m Bool
+handleWebhooks pusher getReq parseReq webhookF =
```

Comment:
IMO we should minimise the number of callbacks these functions take. I think it makes the calling code much clearer. For example I think this function could just take the request ByteStrings directly, and return a `Maybe (UTCTime, WebhookEv)` (or, even better, an `Either`). So essentially just `parseWebhookPayloadReq` but which takes the `Pusher` object for convenience.

If you think this function is worthwhile I'd be interested in seeing an example of a caller using `parseWebhookPayloadReq` (but a version which takes a `Pusher`) and seeing how it compares to the example using the current `handleWebhooks` functions.

I can explain this to you in person if that would help.